ql/src/test/results/clientpositive/beeline/colstats_all_nulls.q.out

```diff
@@ -73,6 +74,7 @@ max_col_len
 num_trues	
 num_falses	
 bit_vector	HL
+histogram	
```

Comment:
The fact that the DESCRIBE FORMATTED output is changing by default makes me think whether it is better to not show this field if histogram stats is disabled.  Since this is brand new (and experimental) feature, the DESCRIBE output could be made conditional on whether it is enabled.  Perhaps this was discussed with other reviewers .. I am not sure.

Reply:
We have two config knobs for enabling histogram/KLL, one is `hive.stats.kll.enable` (for turning it on/off during statistics computation), the other is `metastore.stats.fetch.kll` (which turns on/off the retrieval from the metastore).

If we were to implement this, would you show/hide the **histogram** line for the `DESCRIBE FORMATTED` command based on the value of `metastore.stats.fetch.kll` alone?

I thought of this at some point, but the fact that it hasn't been done for HLL even in the first PR refrained me from implementing it.

The reason is that we would end up having an inconsistent behaviour w.r.t. HLL which is always displayed, no matter what the configuration is for the corresponding `metastore.stats.fetch.bitvector` parameter.

This said, I don't have strong feelings either way and I am fine implementing the hiding mechanism and it wouldn't take much, I just wanted to expose my reasoning so far to have all the pros/cons on the table.

Reply:
> If we were to implement this, would you show/hide the histogram line for the DESCRIBE FORMATTED command based on the value of metastore.stats.fetch.kll alone?

When I made the comment I was not thinking about the metastore flag but yeah, if we were to do this it would make sense to make it conditional on metastore.stats.fetch.kll alone. 
While I understand that displaying it makes it consistent with bit_vector and I see your point about the fact that for bit_vector  it was displayed from the start,  I think that histograms/KLL are more advanced stats and it needs a little bit of time to 'bake' in both in terms of the code and usability.  I am also trying to see if we can reduce the number of q.out changes within a single patch especially since these q.out files have empty histogram stats.  Instead, if we can have a follow up to add this, it would simplify things a bit.  It would also be useful to get additional opinions on this.  @dengzhhu653  @kasakrisz  any thoughts ?

Reply:
I agree on your take, I have committed code making histogram display in `DESCRIBE` based on `metastore.stats.fetch.kll` and updated qout tests (basically reverting what was changed in the first place).